File: ts/tests/indentation.test.ts
import { describe, expect, it } from "vitest"
import { formatters, llml } from "../src/index"

describe("Indentation", () => {
  it("should handle basic formatting (indent option ignored for now)", () => {
    const formatterSet = formatters.vibeXML({ indent: "  " })
    const result = llml({ message: "Hello" }, formatterSet)
    const expected = "<message>Hello</message>"
    expect(result).toBe(expected)
  })

  it("should handle list formatting (indent option ignored for now)", () => {
    const formatterSet = formatters.vibeXML({ indent: "  " })
    const result = llml({ items: ["a", "b"] }, formatterSet)
    const expected = ["<items>", "  <items-1>a</items-1>", "  <items-2>b</items-2>", "</items>"].join("\n")
    expect(result).toBe(expected)
  })
})

File: go/tests/indentation_test.go
package llml_test

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/zenbase-ai/llml/go/pkg/llml"
)

func TestBasicIndentation(t *testing.T) {
	result := llml.Sprintf(map[string]any{
		"message": "Hello",
	}, llml.Options{Indent: "  "})
	expected := "  <message>Hello</message>"
	assert.Equal(t, expected, result)
}

func TestListWithIndentation(t *testing.T) {
	result := llml.Sprintf(map[string]any{
		"items": []any{"a", "b"},
	}, llml.Options{Indent: "  "})
	expected := "  <items-list>\n" +
		"    <items-1>a</items-1>\n" +
		"    <items-2>b</items-2>\n" +
		"  </items-list>"
	assert.Equal(t, expected, result)
}
